### ISSUE TITLE
Rule based uploader fails all datasets if one url has an issue.

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#8488]
I ran the uploader with 6 URLs, one I did not realise was invalid.

All datasets are in error state, but unlike in https://github.com/galaxyproject/galaxy/issues/4021#issuecomment-523518510, they all bear the same error message, so there's no clue which one failed.

```
Traceback (most recent call last):
  File "/opt/galaxy/server/lib/galaxy/tools/data_fetch.py", line 329, in <module>
    main()
  File "/opt/galaxy/server/lib/galaxy/tools/data_fetch.py", line 39, in main
    galaxy_json = _request_to_galaxy_json(upload_config, request)
  File "/opt/galaxy/server/lib/galaxy/tools/data_fetch.py", line 49, in _request_to_galaxy_json
    fetched_target = _fetch_target(upload_config, target)
  File "/opt/galaxy/server/lib/galaxy/tools/data_fetch.py", line 167, in _fetch_target
    elements = elements_tree_map(_resolve_src, items)
  File "/opt/galaxy/server/lib/galaxy/tools/data_fetch.py", line 195, in elements_tree_map
    new_items.append(f(item))
  File "/opt/galaxy/server/lib/galaxy/tools/data_fetch.py", line 96, in _resolve_src
    name, path = _has_src_to_path(upload_config, item, is_dataset=True)
  File "/opt/galaxy/server/lib/galaxy/tools/data_fetch.py", line 223, in _has_src_to_path
    path = sniff.stream_url_to_file(url)
  File "/opt/galaxy/server/lib/galaxy/datatypes/sniff.py", line 57, in stream_url_to_file
    page = urlopen(path)  # page will be .close()ed in stream_to_file
  File "/usr/lib64/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/usr/lib64/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 404: NOT FOUND
```

**Comment by hexylena** [comment]
as comparison:

![image](https://user-images.githubusercontent.com/458683/63447547-829e3900-c43c-11e9-81eb-bb789e3d8ea7.png)

**Comment by assuntad23** [comment]
closed with #11276
